modules/flowable-cmmn-engine/src/test/java/org/flowable/cmmn/test/tenant/ChangeTenantIdCaseInstanceTest.java

```diff
@@ -0,0 +1,264 @@
+    private String deploymentIdWithoutTenant;
+
+    @Before
+    public void setUp() {
+        this.deploymentIdWithTenantA = cmmnRepositoryService.createDeployment()
+                .addClasspathResource("org/flowable/cmmn/test/tenant/caseWithMilestone.cmmn").tenantId(TEST_TENANT_A)
+                .deploy().getId();
+        addDeploymentForAutoCleanup(deploymentIdWithTenantA);
+        this.deploymentIdWithTenantB = cmmnRepositoryService.createDeployment()
+                .addClasspathResource("org/flowable/cmmn/test/tenant/caseWithMilestone.cmmn").tenantId(TEST_TENANT_B)
+                .deploy().getId();
+        addDeploymentForAutoCleanup(deploymentIdWithTenantB);
+        this.deploymentIdWithTenantC = cmmnRepositoryService.createDeployment()
+                .addClasspathResource("org/flowable/cmmn/test/tenant/caseWithMilestone.cmmn").tenantId(TEST_TENANT_C)
+                .deploy().getId();
+        addDeploymentForAutoCleanup(deploymentIdWithTenantC);
+        this.deploymentIdWithoutTenant = cmmnRepositoryService.createDeployment()
+                .addClasspathResource("org/flowable/cmmn/test/tenant/caseWithMilestoneDup.cmmn").deploy().getId();
+        addDeploymentForAutoCleanup(deploymentIdWithoutTenant);
+    }
+
+    @Test
+    public void testDeployments() {
+        assertThat(cmmnRepositoryService.createDeploymentQuery().count()).isEqualTo(4);
+        assertThat(cmmnRepositoryService.createDeploymentQuery().deploymentWithoutTenantId().count()).isEqualTo(1);
+        assertThat(cmmnRepositoryService.createDeploymentQuery().deploymentTenantId(TEST_TENANT_A).count())
+                .isEqualTo(1);
+        assertThat(cmmnRepositoryService.createDeploymentQuery().deploymentTenantId(TEST_TENANT_B).count())
+                .isEqualTo(1);
+        assertThat(cmmnRepositoryService.createDeploymentQuery().deploymentTenantId(TEST_TENANT_C).count())
+                .isEqualTo(1);
+    }
+
+    @Test
+    public void testChangeTenantIdCaseInstance() {
+        // Starting case instances that will be sent to the history
+        String caseInstanceIdACompleted = startCase(TEST_TENANT_A, "caseWithMilestone", "caseInstanceACompleted", true);
+        String caseInstanceIdBCompleted = startCase(TEST_TENANT_B, "caseWithMilestone", "caseInstanceBCompleted", true);
+        String caseInstanceIdCCompleted = startCase(TEST_TENANT_C, "caseWithMilestone", "caseInstanceCCompleted", true);
+        
+        // Starting case instances that will be kept active
+        String caseInstanceIdAActive = startCase(TEST_TENANT_A, "caseWithMilestone", "caseInstanceAActive", false);
+        String caseInstanceIdBActive = startCase(TEST_TENANT_B, "caseWithMilestone", "caseInstanceBActive", false);
+        String caseInstanceIdCActive = startCase(TEST_TENANT_C, "caseWithMilestone", "caseInstanceCActive", false);
+        
+        Set<String> caseInstanceIdsTenantA = Sets.newSet(caseInstanceIdACompleted, caseInstanceIdAActive);
```

Comment:
Use `new HashSet<>(Arrays.asList(caseInstanceIdACompleted, caseInstanceIdAActive")` instead of using `Sets` from an internal Mockito API

Reply:
Done